src/Components/Components/src/Properties/ILLink.Substitutions.xml

```diff
@@ -0,0 +1,12 @@
+<linker>
+  <assembly fullname="Microsoft.AspNetCore.Components" >
+    <!-- Trim away hot-reload -->
+    <type fullname="Microsoft.AspNetCore.Components.HotReload.HotReloadEnvironment">
```

Comment:
Do we need/want a `feature` value here? Or is the idea that whenever you are trimming this assembly, turn off HotReload?

Reply:
> Do you foresee any reason to put this behind a switch?

No, not at this time. I was just making sure my assumption was correct. Maybe sticking a quick comment in the xml file would help.

Just an FYI - in `dotnet/runtime` we also run the trimmer on each shared fx assembly during the build in order to trim away any dead code (which can be there because of shared source files), and in System.Private.CoreLib it will trim the IL for specific OS and architecture, since CoreLib targets each separately. If you were to ever run the trimmer in this way, this file would still take affect and it would trim away the HotReload stuff during your build. I don't think you have plans to do this, but I just wanted to call it out for informational purposes.

Reply:
We do want these types to be present when users are developing, so we couldn't trim it as part of our build. Our codebase has fewer platform specific implementations (DataProtection is the only one that comes to mind which has lots of windows specific code).

Reply:
FYI @davidfowl